core/src/main/java/io/questdb/griffin/engine/functions/groupby/RegressionInterceptFunctionFactory.java

```diff
@@ -0,0 +1,230 @@
+                double var = sumX / count;
+
+                double slope = covar / var;
+
+                double meanY = rec.getDouble(valueIndex);
+                double meanX = rec.getDouble(valueIndex + 1);
+
+                return meanY - meanX * slope;
+            }
+            return Double.NaN;
+        }
+
+        @Override
+        public Function getLeft() {
+            return yFunc;
+        }
+
+        @Override
+        public String getName() {
+            return "regr_intercept";
+        }
+
+        @Override
+        public Function getRight() {
+            return xFunc;
+        }
+
+        @Override
+        public int getValueIndex() {
+            return valueIndex;
+        }
+
+        @Override
+        public void initValueIndex(int valueIndex) {
+            this.valueIndex = valueIndex;
+        }
+
+        @Override
+        public void initValueTypes(ArrayColumnTypes columnTypes) {
+            this.valueIndex = columnTypes.getColumnCount();
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.LONG);
+        }
+
+        @Override
+        public boolean isConstant() {
+            return false;
+        }
+
+        @Override
+        public boolean isThreadSafe() {
+            return BinaryFunction.super.isThreadSafe();
+        }
+
+        @Override
+        public void merge(MapValue destValue, MapValue srcValue) {
+            double srcMeanY = srcValue.getDouble(valueIndex);
+            double srcMeanX = srcValue.getDouble(valueIndex + 1);
+            double srcSumXY = srcValue.getDouble(valueIndex + 2);
+            double srcSumX = srcValue.getDouble(valueIndex + 3);
+            long srcCount = srcValue.getLong(valueIndex + 4);
+
+            double destMeanY = destValue.getDouble(valueIndex);
+            double destMeanX = destValue.getDouble(valueIndex + 1);
+            double destSumXY = destValue.getDouble(valueIndex + 2);
+            double destSumX = destValue.getDouble(valueIndex + 3);
+            long destCount = destValue.getLong(valueIndex + 4);
+
+            long mergedCount = srcCount + destCount;
+            double deltaY = destMeanY - srcMeanY;
+            double deltaX = destMeanX - srcMeanX;
+
+            // This is only valid when countA is much larger than countB.
+            // If both are large and similar sizes, delta is not scaled down.
+            // double mergedMean = srcMean + delta * ((double) destCount / mergedCount);
+
+            // So we use this instead:
+            double mergedMeanY = (srcCount * srcMeanY + destCount * destMeanY) / mergedCount;
+            double mergedMeanX = (srcCount * srcMeanX + destCount * destMeanX) / mergedCount;
+            double mergedSumXY = srcSumXY + destSumXY + (deltaY * deltaX) * ((double) (srcCount * destCount) / mergedCount);
```

Comment:
this is a potential numeric overflow that can be easily avoided:
```suggestion
            double mergedSumXY = srcSumXY + destSumXY + (deltaY * deltaX) * (((double) srcCount * destCount) / mergedCount);
```